src/systems/systems/src/pressurization/pressure_valve.rs

```diff
@@ -1,44 +1,99 @@
+    pub fn new_outflow_valve() -> Self {
         Self {
             open_amount: Ratio::new::<percent>(100.),
             target_open: Ratio::new::<percent>(100.),
             full_travel_time: Duration::from_secs(4),
+            manual_travel_time: Duration::from_secs(60),
         }
     }
 
-    pub fn update<T: PressureValveActuator>(&mut self, context: &UpdateContext, actuator: &T) {
-        self.target_open = actuator.target_valve_position();
-        if self.open_amount < self.target_open {
-            self.open_amount += Ratio::new::<percent>(
-                self.get_valve_change_for_delta(context)
-                    .min(self.target_open.get::<percent>() - self.open_amount.get::<percent>()),
-            );
-        } else if self.open_amount > self.target_open {
-            self.open_amount -= Ratio::new::<percent>(
-                self.get_valve_change_for_delta(context)
-                    .min(self.open_amount.get::<percent>() - self.target_open.get::<percent>()),
-            );
+    pub fn new_safety_valve() -> Self {
+        Self {
+            open_amount: Ratio::new::<percent>(0.),
+            target_open: Ratio::new::<percent>(0.),
+            full_travel_time: Duration::from_secs(1),
+            manual_travel_time: Duration::from_secs(1),
+        }
+    }
+
+    pub fn update(
+        &mut self,
+        context: &UpdateContext,
+        signal: &impl ControllerSignal<PressureValveSignal>,
+    ) {
+        match signal.signal() {
+            Some(PressureValveSignal::Open) => {
+                if self.open_amount < Ratio::new::<percent>(100.) {
+                    self.open_amount += Ratio::new::<percent>(
+                        self.get_manual_change_for_delta(context)
+                            .min(100. - self.open_amount.get::<percent>()),
+                    );
+                }
+            }
+            Some(PressureValveSignal::Close) => {
+                if self.open_amount > Ratio::new::<percent>(0.) {
+                    self.open_amount -= Ratio::new::<percent>(
+                        self.get_manual_change_for_delta(context)
+                            .min(self.open_amount.get::<percent>()),
+                    );
+                }
+            }
+            Some(PressureValveSignal::Neutral) => (),
+            _ => {
+                if self.open_amount < self.target_open {
+                    self.open_amount +=
+                        Ratio::new::<percent>(self.get_valve_change_for_delta(context).min(
+                            self.target_open.get::<percent>() - self.open_amount.get::<percent>(),
+                        ));
+                } else if self.open_amount > self.target_open {
+                    self.open_amount -=
+                        Ratio::new::<percent>(self.get_valve_change_for_delta(context).min(
+                            self.open_amount.get::<percent>() - self.target_open.get::<percent>(),
+                        ));
+                }
```

Comment:
**question**: For my understanding. This code is only used when manual mode is on and only in manual mode does the valve not fully open or close all the time? It's interesting that in automatic mode the valve always opens fully and in manual mode it doesn't. 😀

Reply:
No, it's the opposite 😄. As explained a few comments above, this part of the code only triggers when both the `ResidualPressureController` and the `PressurizationOverheadPanel` are sending a `None` signal (i.e., not active and not in manual mode, respectively).

When this happens, the `self.target_open` comes from `cpc.target_valve_position()`